api/datastore/sql/migrations/migrations.go

```diff
@@ -55,6 +55,35 @@ var initialTables = [...]string{`CREATE TABLE IF NOT EXISTS routes (
 );`,
 }
 
+func checkOldMigrationTableVersionIfExists(db *sql.DB) (version int64, dirty bool, err error) {
+	migrationsTable := "schema_migrations"
+	ctx := context.Background()
+
+	// deleting old table after all
+	defer func() {
+		_, _ = db.ExecContext(ctx, "DROP TABLE IF EXISTS "+migrationsTable)
```

Comment:
this needs to happen after we set the version in the new migrations table only? otherwise if migrations fail, we risk skipping them.

Reply:
fixed